### PR TITLE
HTTPClient fixes for EOF read, chunked transfer encoding

### PR DESCRIPTION
**Pull request by Faless**
Fix HTTPClient `Connection: close` mode (was disconnecting when the server held back responding like when using event stream). Closes #25985 .

Fix HTTPClient keep alive with chunked encoding.

We need to consume the trailer part and final CRLF after last chunk as per [RFC 7230 section 4.1](https://tools.ietf.org/html/rfc7230#section-4.1):

```
chunked-body   = *chunk
                 last-chunk
                 trailer-part
                 CRLF
```

We do not return the trailer part, just consume it allowing following requests to work as expected when using keep alive.

### CODE REVIEW

**Review by RandomShaper — APPROVED** [review]
I can't see anything wrong and the code is simpler now, but it's hard to follow mentally.

We desperately need automated testing...

**Comment by akien-mga** [comment]
Thanks!

**Comment by Faless** [comment]
@RandomShaper I made them few months ago (I plan to change HTTPClient to use an external parser which I can also use for server).
Have a look at https://github.com/Faless/gd-tests and let me know what you think

**Comment by RandomShaper** [comment]
Now that you mention it, I remember you talked about it during the Sprint. :)
